mlflow/models/python_api.py

```diff
@@ -221,54 +268,36 @@ def _serialize_input_data(input_data, content_type):
-        return input_data
+    if content_type == _CONTENT_TYPE_CSV:
+        if isinstance(input_data, str):
+            _validate_string(input_data, content_type)
+            return input_data
+        else:
+            try:
+                return pd.DataFrame(input_data).to_csv(index=False)
+            except Exception as e:
+                raise MlflowException.invalid_parameter_value(
+                    "Failed to serialize input data to CSV format."
+                ) from e
 
     try:
-        if content_type == _CONTENT_TYPE_CSV:
-            # We should not set header=False here because the scoring server expects the
-            # first row to be the header
-            return pd.DataFrame(input_data).to_csv(index=False)
-        else:
-            return _serialize_to_json(input_data)
+        # we do not need to validate string input here
+        # as the model can accept arbitrary string inputs
+        return convert_input_example_to_serving_input(input_data)
     except Exception as e:
         raise MlflowException.invalid_parameter_value(
-            message=f"Input data could not be serialized to {content_type}."
+            "Invalid input data, please make sure the data is acceptable by the "
+            "loaded pyfunc model. Use `mlflow.models.convert_input_example_to_serving_input` "
+            "to manually validate your input data."
         ) from e
 
 
-def _serialize_to_json(input_data):
-    # imported inside function to avoid circular import
-    from mlflow.pyfunc.scoring_server import SUPPORTED_FORMATS
-    from mlflow.pyfunc.utils.serving_data_parser import SUPPORTED_LLM_FORMATS
-
-    if isinstance(input_data, dict) and any(
-        key in input_data for key in SUPPORTED_FORMATS | SUPPORTED_LLM_FORMATS
-    ):
-        return json.dumps(input_data)
-    else:
-        original_input_data = input_data
-
-        if isinstance(input_data, (datetime, bool, bytes, float, int, str)):
-            input_data = [input_data]
-
-        input_data = dump_input_data(input_data)
-
-        _logger.info(
-            f"Your input data has been transformed to comply with the expected input format for "
-            "the MLflow scoring server. If you want to deploy the model to online serving, make "
-            "sure to apply the same preprocessing in your inference client. Please also refer to "
-            "https://www.mlflow.org/docs/latest/deployment/deploy-model-locally.html#json-input "
-            "for more details on the supported input format."
-            f"\n\nOriginal input data:\n{original_input_data}"
-            f"\n\nTransformed input data:\n{input_data}"
-        )
-        return input_data
-
-
 def _validate_string(input_data: str, content_type: str):
+    """
+    Validate the input data is compatible with the content_type.
+    If content_type is 'csv', the string must be the path to a CSV file.
+    If content_type is 'json', the string must be a valid JSON string.
```

Comment:
It seems we only use this function for CSV content type, can we remove JSON-related piece here?